api/server.py

```diff
@@ -378,7 +379,128 @@ def download_crash_id(crash_id):
+
+    if request.method == "GET":
+        # get filename from DB
+        res = make_hasura_request(
+            query=GET_PERSON_IMAGE_METADATA, variables={"person_id": safe_person_id}
+        )
+        app.logger.info(res)
+        obj_key = res["people_by_pk"]["image_s3_object_key"]
+
+        if not obj_key:
+            return jsonify(error=f"No image found for person ID: {safe_person_id}"), 404
+
+        url = s3.generate_presigned_url(
+            ExpiresIn=3600,
+            ClientMethod="get_object",
+            Params={
+                "Bucket": AWS_S3_BUCKET,
+                "Key": obj_key,
+            },
+        )
+        return jsonify(url=url)
+
+    elif request.method == "POST":
+        if "file" not in request.files:
+            return jsonify(error="No file provided"), 400
+
+        file = request.files["file"]
+
+        if file.filename == "":
+            return jsonify(error="No file selected"), 400
+
+        try:
+            # validate image file
+            file.seek(0)
+            img = Image.open(file)
+            img.verify()
+
+            # get actual image format
+            file.seek(0)
+            img = Image.open(file)
+            actual_format = img.format.lower()
+
+            allowed_formats = ["jpeg", "png"]
+            if actual_format not in allowed_formats:
+                return jsonify(error=f"Image format must be JPEG or PNG"), 400
+
+            app.logger.info(f"Validated image format: {actual_format}")
+
+            # todo: do we want this?
+            width, height = img.size
+            if width > MAX_IMAGE_PIXELS or height > MAX_IMAGE_PIXELS:
+                return jsonify(error="Image dimensions too large"), 400
+
+            # reset file position for upload
+            file.seek(0)
+
+            ext = "jpg" if actual_format == "jpeg" else actual_format
+
+        except Exception as e:
+            app.logger.error(f"Image validation failed: {e}")
+            return jsonify(error="Invalid or corrupted image file"), 400
+
+        # prepare for S3 upload
+        filename = f"{safe_person_id}.{ext}"
+        obj_key = f"{AWS_S3_PERSON_IMAGE_LOCATION}/{filename}"
+        app.logger.info(f"Uploading image: {AWS_S3_BUCKET}/{obj_key}")
+
+        try:
+            s3.upload_fileobj(
+                file,
+                AWS_S3_BUCKET,
+                obj_key,
+                ExtraArgs={
+                    "ContentType": file.content_type or "image/jpeg",
+                },
+            )
+
+        except Exception as e:
+            app.logger.exception(str(e))
+            return jsonify(error=f"Upload failed"), 500
+
+        try:
+            res = make_hasura_request(
+                query=UPDATE_PERSON_IMAGE_METADATA,
+                variables={
+                    "person_id": safe_person_id,
+                    "image_s3_object_key": obj_key,
+                    "updated_by": get_user_email(),
```

Comment:
here's where we make use of our new [user email claim](https://austininnovation.slack.com/archives/CM9MK950S/p1765916319698789) on the JWT.